src/github.com/mozilla-services/pushgo/simplepush/handlers_endpoint.go

```diff
@@ -337,16 +339,22 @@ func (h *EndpointHandler) deliver(cn http.CloseNotifier, uaid, chid string,
 			cancelSignal = cn.CloseNotify()
 		}
 		// Route the update.
+		startTime := timeNow().UTC()
 		delivered, _ = h.router.Route(cancelSignal, uaid, chid, version,
-			timeNow().UTC(), requestID, data)
+			startTime, requestID, data)
+		routingTime = timeNow().UTC().Sub(startTime)
 		if delivered {
+			// Routing succeeded.
+			h.metrics.Increment("router.broadcast.hit")
+			h.metrics.Timer("updates.routed.hits", routingTime)
 			return true
 		}
 	}
-	// If the device is not connected to this server, indicate whether routing
-	// was successful.
 	if !workerConnected {
-		return
+		// Device is not connected to this server and routing failed.
+		h.metrics.Increment("router.broadcast.miss")
+		h.metrics.Timer("updates.routed.misses", routingTime)
```

Comment:
I don't think so...if `AlwaysRoute` is enabled or the client isn't connected to us, broadcast the message. If another node accepts the update, increment `router.broadcast.hit` and return. Otherwise, if the client isn't connected (to catch the case where `AlwaysRoute` is enabled but the client is connected to us), increment `router.broadcast.miss` and return. Finally, we know the client is connected to us, so try local delivery.

Would adding a `useRouting` variable be clearer?